### PR TITLE
Misc Updates to changelog for 6.0

### PR DESCRIPTION
**Pull request by Carreau**
Make it slightly easier for users to understand why they shoudl migrate
(or not) to 6.0. Feel free to add entries.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+
+IPykernel 6 should contain all changes of the 5.x series, in addition to the
+followings non-exhaustive changes.
+
+
+ - The control channel on IPykernel 6.0 is ran in a separate thread, this may
+   change the order in which messages are process, though this change was necessary
+   to accommodate the debugger.
+
+ - Support for the debugger protocol, when using jupyter lab, retrolab or any
```
Comment (blink1073) [inline CHANGELOG.md:22]:
```suggestion
 - Support for the debugger protocol, when using `JupyterLab`, `RetroLab` or any
```

Comment (blink1073) [inline CHANGELOG.md:22]: Maybe swap this bullet with the first to lead with the feature?

**Review by blink1073 — APPROVED** [review]
Thanks!  Just a couple suggestions

File: CHANGELOG.md (hunk 1 above)
Comment (Carreau) [reply CHANGELOG.md]: Thanks, done and added a couple other things. I'll probably send more updates in subsequent PRs there are still a couple of deprecations to remove.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,59 @@
 # Changes in IPython kernel
 
-* remove `find_connection_file` and profile arg of `connect_qtconsole` and `get_connection_info`, deprecated since IPykernel 4.2.2.
+
+## 6.0.0
+
+IPykernel 6.0 is the first major release in about two years, that brings a number of improvements code cleanup and new
```
Comment (willingc) [inline CHANGELOG.md:6]:
```suggestion
IPykernel 6.0 is the first major release in about two years, that brings a number of improvements, code cleanup, and new
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,59 @@
...
+features to IPython.
+
+You should be able to view all closed issues and merged Pull Request for this
+milestone [on
+GitHub](https://github.com/ipython/ipykernel/issues?q=milestone%3A6.0+is%3Aclosed+),
+as for any major releases, we advise greater care when updating that for minor
+release and welcome any feedback (~50 Pull-requests).
+
+IPykernel 6 should contain all changes of the 5.x series, in addition to the
+followings non-exhaustive changes.
```
Comment (willingc) [inline CHANGELOG.md:16]:
```suggestion
following non-exhaustive changes.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,59 @@
...
+
+
+ - Support for the debugger protocol, when using `JupyterLab`, `RetroLab` or any
+   frontend supporting the debugger protocol you should have access to the
+   debugger functionalities.
+
+ - The control channel on IPykernel 6.0 is ran in a separate thread, this may
```
Comment (willingc) [inline CHANGELOG.md:23]:
```suggestion
 - The control channel on IPykernel 6.0 is run in a separate thread, this may
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,59 @@
...
+   change the order in which messages are process, though this change was necessary
```
Comment (willingc) [inline CHANGELOG.md:24]:
```suggestion
   change the order in which messages are processed, though this change was necessary
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,59 @@
...
+## Deprecations in 6.0
+
+ - `Kernel`s now support only a single shell stream, multiple streams will now be ignored. The attribute
+   `Kernel.shell_streams` (plural) is deprecated in ipykernel 6.0. Use `Kernel.shell_stream` (singular)
+
+ - `Kernel._parent_header` is deprecated, even though it was private. Use `.get_parent()` now.
+
+## Removal in 6.0
+
+ - ipykernel.codeutils was deprecated since 4.x series (2016) and has been removed, please import similar
```
Comment (willingc) [inline CHANGELOG.md:52]:
```suggestion
 - `ipykernel.codeutils` was deprecated since 4.x series (2016) and has been removed, please import similar
```

**Review by willingc — APPROVED** [review]
Thanks @Carreau :smile: I've left a few small typo corrections. Otherwise, this is good to merge.

**Comment by Carreau** [comment]
Thanks @willingc  !

**Pull request merged**